dcu

```diff
@@ -360,6 +448,175 @@ dcu_logo_command() {
 
 }
 
+# :command.function
+dcu_variable_command() {
+  # src/variable_command.sh
+  # SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
+  #
+  # SPDX-License-Identifier: Apache-2.0
+
+  # inspect_args
+
+  DASHARO_ROM="${args[dasharo_rom_file]}"
+  SMMSTORETOOL="smmstoretool"
+  GET="${args[--get]}"
+  VALUE="${args[--value]}"
+  SET="${args[--set]}"
+  LIST="${args[--list]}"
+
+  typeof()
+  {
+    # TODO: determine type programatically
+    case $1 in
+      LockBios \
+      |NetworkBoot \
+      |UsbDriverStack \
+      |UsbMassStorage \
+      |SmmBwp \
+      |Ps2Controller \
+      |BootManagerEnabled \
+      |PCIeResizeableBarsEnabled \
+      |EnableCamera \
+      |EnableWifiBt \
+      |SerialRedirection \
+      |SerialRedirection2)
+        echo "bool"
+        ;;
+      MeMode)
+        echo "memode"
+        ;;
+      FanCurveOption)
+        echo "fancurve"
+        ;;
+      *)
+        echo "unknown"
+        ;;
+    esac
+  }
+
+  valueof()
+  {
+    case `typeof $1` in
+      bool)
+        if [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t bool` = "false" ]; then
+          echo "Disabled"
+        elif [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t bool` = "true" ]; then
+          echo "Enabled"
+        else
+          echo "Error!"
+          exit 1
+        fi
+        ;;
+      memode)
+        if [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t uint8` = "0" ]; then
+          echo "Enabled"
+        elif [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t uint8` = "1" ]; then
+          echo "DisabledSoft"
+        elif [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t uint8` = "2" ]; then
+          echo "DisabledHAP"
+        else
+          echo "Error!"
+          exit 1
+        fi
+        ;;
+      fancurve)
+        if [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t uint8` = "0" ]; then
+          echo "Silent"
+        elif [ `${SMMSTORETOOL} ${DASHARO_ROM} get -g dasharo -n $1 -t uint8` = "1" ]; then
+          echo "Performance"
+        else
+          echo "Error!"
+          exit 1
+        fi
+        ;;
+    esac
+  }
+
+  acceptedvaluesfor()
+  {
+    case `typeof $1` in
+      bool)
+        echo "Enabled / Disabled"
+        ;;
+      memode)
+        echo "Enabled / DisabledSoft / DisabledHAP"
+        ;;
+      fancurve)
+        echo "Silent / Performance"
+        ;;
+      *)
+        ;;
+    esac
+  }
+
+  get_variable()
+  {
+    echo $(valueof ${GET})
+  }
+
+  set_variable()
+  {
+    if [ -z "${VALUE}" ]
+    then
+      echo "Value to set not provided, exiting" >&2
+      exit 1
+    fi
+    if [ -z "$(acceptedvaluesfor ${SET})" ]; then
+      echo "Setting ${SET} is not supported"
+      exit 1
```

Comment:
These should be changed to something not conflicting with already used exitcodes

Reply:
https://github.com/Dasharo/dcu/pull/5/commits/4c9d0abdd4cf1defeaacd5fcff2a43bd16c81e69